src/main/java/com/shapesecurity/salvation/data/Policy.java

```diff
@@ -601,33 +694,60 @@ public boolean allowsStyleWithHash(@Nonnull HashAlgorithm algorithm, @Nonnull Ba
 		if (this.allowsUnsafeInlineStyle()) {
 			return true;
 		}
-		StyleSrcDirective styleSrcDirective = this.getDirectiveByType(StyleSrcDirective.class);
-		if (styleSrcDirective == null) {
-			return this.defaultsAllowHash(algorithm, hashValue);
+		StyleSrcElemDirective styleSrcElemDirective = this.getDirectiveByType(StyleSrcElemDirective.class);
+		if (styleSrcElemDirective == null) {
```

Comment:
This kind of nesting is a bit weird. I'd prefer to do a `!=` check with an early return so that our locals are all kept as contiguous as possible.

```java
StyleSrcElemDirective styleSrcElemDirective = this.getDirectiveByType(StyleSrcElemDirective.class);
if (styleSrcElemDirective != null) {
	return styleSrcElemDirective.matchesHash(algorithm, hashValue);
}

StyleSrcDirective styleSrcDirective = this.getDirectiveByType(StyleSrcDirective.class);
if (styleSrcDirective != null) {
	return styleSrcDirective.matchesHash(algorithm, hashValue);
}

return this.defaultsAllowHash(algorithm, hashValue);
```